File: src/main/java/bruteforce/boj_2231/Main.java
package bruteforce.boj_2231;

import java.io.*;

// 74784kb 284ms
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());
        int result = 0;

        for(int i=1; i<=n;i++) {
            int sum = i;
            for(char c: String.valueOf(i).toCharArray()) {
                sum += Character.getNumericValue(c);
            };
            if (sum == n) {
                result = i;
                break;
            }
        }
        bw.write(result + "");
        bw.flush();
        bw.close();
        br.close();
    }
}

File: src/main/java/bruteforce/boj_2231/Main.kt
package bruteforce.boj_2231

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

// 12756kb 104ms
fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val input = br.readLine().toInt()
    var result = 0

    for (i in 1..input) {
        var sum = i
        var n = i
        while(n > 0) {
            sum += n % 10
            n /= 10
        }
        if (sum == input) {
            result = i
            break
        }
    }

    bw.write("$result")
    bw.flush()
    bw.close()
    br.close()
}